FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/SchedulerService.cs

```diff
@@ -0,0 +1,398 @@
+            {
+                TriggerLeaseEntity triggerLeaseEntity;
+                var needRenewLease = true;
+                try
+                {
+                    // try to get trigger lease entity
+                    triggerLeaseEntity = await _metadataStore.GetTriggerLeaseEntityAsync(_queueType, cancellationToken);
+                }
+                catch (RequestFailedException ex) when (ex.ErrorCode == AzureStorageErrorCode.GetEntityNotFoundErrorCode)
+                {
+                    _logger.LogWarning("The current trigger lease doesn't exist, will create a new one.");
+
+                    var initialTriggerLeaseEntity = new TriggerLeaseEntity()
+                    {
+                        PartitionKey = TableKeyProvider.LeasePartitionKey(_queueType),
+                        RowKey = TableKeyProvider.LeaseRowKey(_queueType),
+                        WorkingInstanceGuid = _instanceGuid,
+                        HeartbeatDateTime = DateTimeOffset.UtcNow,
+                    };
+
+                    // add the initial trigger entity to table
+                    await _metadataStore.AddEntityAsync(initialTriggerLeaseEntity, cancellationToken);
+
+                    // try to get trigger lease entity after insert
+                    triggerLeaseEntity = await _metadataStore.GetTriggerLeaseEntityAsync(_queueType, cancellationToken);
+                    needRenewLease = false;
+                }
+
+                if (triggerLeaseEntity == null)
+                {
+                    _logger.LogWarning("Try to acquire lease failed, the retrieved lease trigger entity is null.");
+                    return false;
+                }
+
+                if (triggerLeaseEntity.WorkingInstanceGuid != _instanceGuid &&
+                    triggerLeaseEntity.HeartbeatDateTime.AddSeconds(SchedulerServiceLeaseExpirationInSeconds) > DateTimeOffset.UtcNow)
+                {
+                    _logger.LogInformation("Try to acquire lease failed, another worker is using it.");
+                    return false;
+                }
+
+                if (needRenewLease)
+                {
+                    triggerLeaseEntity.WorkingInstanceGuid = _instanceGuid;
+                    triggerLeaseEntity.HeartbeatDateTime = DateTimeOffset.UtcNow;
+
+                    await _metadataStore.UpdateEntityAsync(triggerLeaseEntity, cancellationToken);
+                }
+            }
+            catch (Exception ex)
+            {
+                _logger.LogWarning(
+                    $"Try acquire lease failed. Exception: {ex.Message}.");
+                return false;
+            }
+
+            _logger.LogInformation("Acquire lease successfully.");
+            return true;
+        }
+
+        private async Task<bool> TryPullAndProcessTriggerEntity(CancellationToken cancellationToken)
+        {
+            // try next time if the queue client isn't initialized.
+            if (!_queueClient.IsInitialized())
```

Comment:
Why it is possible the queue not initialzied/

Reply:
or we can put it at scheduler start?

Reply:
Fixed